tests/benchmark/s2n_neg_server_benchmark.cc

```diff
@@ -0,0 +1,899 @@
+    }
+
+    if (listen(sockfd, 1) == -1) {
+        fprintf(stderr, "listen error: %s\n", strerror(errno));
+        exit(1);
+    }
+
+    if (DEBUG_PRINT)
+        printf("Listening on %s:%s\n", host, port);
+
+    struct s2n_config *config = s2n_config_new();
+    if (!config) {
+        print_s2n_error("Error getting new s2n config");
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_config_set_server_max_early_data_size(config, max_early_data),
+               "Error setting max early data");
+
+    GUARD_EXIT(s2n_config_add_dhparams(config, dhparams), "Error adding DH parameters");
+
+    GUARD_EXIT(s2n_config_set_cipher_preferences(config, cipher_prefs), "Error setting cipher prefs");
+
+    GUARD_EXIT(s2n_config_set_cache_store_callback(config, cache_store_callback, session_cache),
+               "Error setting cache store callback");
+
+    GUARD_EXIT(s2n_config_set_cache_retrieve_callback(config, cache_retrieve_callback, session_cache),
+               "Error setting cache retrieve callback");
+
+    GUARD_EXIT(s2n_config_set_cache_delete_callback(config, cache_delete_callback, session_cache),
+               "Error setting cache retrieve callback");
+
+    if (conn_settings.enable_mfl) {
+        GUARD_EXIT(s2n_config_accept_max_fragment_length(config),
+                   "Error enabling TLS maximum fragment length extension in server");
+    }
+
+    if (s2n_config_set_verify_host_callback(config, unsafe_verify_host_fn, NULL)) {
+        print_s2n_error("Failure to set hostname verification callback");
+        exit(1);
+    }
+
+    if (conn_settings.session_ticket) {
+        GUARD_EXIT(s2n_config_set_session_tickets_onoff(config, 1), "Error enabling session tickets");
+    }
+
+    if (conn_settings.session_cache) {
+        GUARD_EXIT(s2n_config_set_session_cache_onoff(config, 1), "Error enabling session cache using id");
+    }
+
+    if (conn_settings.session_ticket || conn_settings.session_cache) {
+        /* Key initialization */
+        uint8_t *st_key;
+        uint32_t st_key_length;
+
+        if (session_ticket_key_file_path) {
+            int fd = open(session_ticket_key_file_path, O_RDONLY);
+            GUARD_EXIT(fd, "Error opening session ticket key file");
+
+            struct stat st;
+            GUARD_EXIT(fstat(fd, &st), "Error fstat-ing session ticket key file");
+
+            st_key = (uint8_t *) mmap(0, st.st_size, PROT_READ, MAP_PRIVATE, fd, 0);
+            st_key_length = st.st_size;
+
+            close(fd);
+        } else {
+            st_key = default_ticket_key;
+            st_key_length = sizeof(default_ticket_key);
+        }
+
+        if (s2n_config_add_ticket_crypto_key(config, ticket_key_name, strlen((char *) ticket_key_name), st_key,
+                                             st_key_length, 0) != 0) {
+            fprintf(stderr, "Error adding ticket key: '%s'\n", s2n_strerror(s2n_errno, "EN"));
+            exit(1);
+        }
+    }
+
+    for (int j = 0; j < 36 ; ++j) {
```

Comment:
2 things:
1. instead of hard coding 36 can this use the list of cipher suites and calculate the size?
2. Why does the server need to count the cipher suites? I thought we start it once and let it run forever?